### DISCUSSION TITLE
URL Encoding - %20 vs Dashes

### DISCUSSION THREAD

**Discussion by danbush** [squidfunk/mkdocs-material#2214] [Q&A] [answered]
Hello! I notice that both mkdocs own site, as well as mkdocs-material’s own site appear to replace each space with a dash in the urls, but the default mkdocs-material experience seems to be encoding spaces with %20. Does anyone know how I can make my site use dashes?

**Comment by squidfunk** [comment]
In which URLs are you seeing the `%20`? Could you provide more details and a minimal reproducible test case?

**Comment by danbush** [comment, marked as answer]
Sure. For clarity: your own documentation site, as well as mkdocs uses dashes as expected, but my build appears to use %20. Full disclosure, I'm using this to create docs for a very stupid DND campaign, but here's a [direct link example](http://dailygrind.crownoffire.ltd/People/Player%20Characters/Davric%20Greenleaf/) (or in plaintext `http://dailygrind.crownoffire.ltd/People/Player%20Characters/Davric%20Greenleaf/`). It's totally possible that something is just not set up right on my end, though I'm not sure what that would be.

**Reply by squidfunk** [reply]
Do you have spaces in your Markdown file names? If so, you should replace them with dashes. For reference, see the setup guide folder, which lists all files with dashes:

https://github.com/squidfunk/mkdocs-material/tree/master/docs/setup

As I understand, Markdown filenames are not slugified, only anchor links.

**Reply by danbush** [reply]
Oh interesting, that’s it. Thank you!
